__tests__/app.test.js

```diff
@@ -258,10 +258,68 @@ describe("GET /api/articles", () => {
     })
     test("returns '404 - path not found' if URL incorrect", () => {
         return request(app)
-            .get("/api/toepics")
+            .get("/api/shrel")
             .expect(404)
             .then(({ body }) => {
                 expect(body.msg).toBe("path not found")
             }) 
     })
+})
+
+describe("GET /api/articles/:article_id/comments", () => {
+    test("Responds with an array of comments for the given article_id of which each comment should have have a list of specified properties.", () => {
+        return request(app)
+            .get("/api/articles/9/comments")
+            .expect(200)
+            .then((res) => {
+                const comments = res.body.comments;
+                expect(comments).toBeInstanceOf(Array);
+                expect(comments.length).toBe(2);
+                comments.forEach(comment => {
+                    expect(comment).toEqual(
+                        expect.objectContaining({
+                            comment_id: expect.any(Number),
+                            votes: expect.any(Number),
+                            created_at: expect.any(String),
+                            author: expect.any(String),
+                            body: expect.any(String),
+                        })
+                    )
+                }) 
+            })
+    })
+    test("Responds with an empty array if there are no comments for the article id", () => {
+        return request(app)
+            .get("/api/articles/4/comments")
+            .expect(200)
+            .then((res) => {
+                const comments = res.body.comments;
+                expect(comments).toBeInstanceOf(Array);
+                expect(comments.length).toBe(0);
+            })
+    })
+    test("returns '404 - path not found' if URL incorrect", () => {
+        return request(app)
+            .get("/api/articles/:article_id/shrel")
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("path not found")
+            }) 
+    })
+    test("returns '404 - article not found' if id doesn't exist", () => {
+        return request(app)
+            .get("/api/articles/1000")
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("article not found")
+            }) 
+    })
+    test("returns '400 - bad request' if id type is wrong", () => {
+        return request(app)
+            .get("/api/articles/one")
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("bad request")
+            }) 
+    })
```

Comment:
I think both of these tests ought to be going to `api/articles/:article_id/comments` - you missed off the `comments` at the end of both! As a result there might be a little more work to do with error handling.

Otherwise - looking good!

Reply:
Ah yeah, a few copy paste jobs that have caught me out, will get onto them now, thanks Liam :)